cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -248,12 +232,12 @@ constexpr T parse_numeric(char const* begin,
       if (*begin == '-' || *begin == '+') { ++begin; }
       int32_t exponent = 0;
       while (begin < end) {
-        exponent = (exponent * 10) + decode_digit<T>(*(begin++), &all_digits_valid);
+        exponent = (exponent * 10) + decode_digit<T, as_hex>(*(begin++), &all_digits_valid);
       }
       if (exponent != 0) { value *= exp10(double(exponent * exponent_sign)); }
     }
   }
-  if (!all_digits_valid) { return error_result; }
+  if (!all_digits_valid) { return {}; }
```

Comment:
Let's be more explicit here. Current code could be mistaken for returning a zero.
```suggestion
  if (!all_digits_valid) { return std::nullopt; }
```

Reply:
Compiles locally, but CI build gives this error 
`04:37:13 $SRC_DIR/cpp/src/io/utilities/parsing_utils.cuh(240): error #20014-D: calling a __host__ function from a __host__ __device__ function is not allowed`

The constructor from nullopt_t arg might be a host function. not constexpr.

Reply:
@karthikeyann Hi,I also encountered this problem, did you solve this problem?

Reply:
Yes. By callling default constructor of std::optional
`if (!all_digits_valid) { return std::optional<T>{}; }`